cmd/booster-http/gateway_handler.go

```diff
@@ -52,34 +58,61 @@ func (h *gatewayHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if h.nitroRpcClient != nil {
+		// This the payment we expect to receive for the file.
+		const expectedPayment = int64(5)
 
 		params, _ := url.ParseQuery(r.URL.RawQuery)
-		if !params.Has("channelId") {
-			webError(w, fmt.Errorf("a valid channel id must be provided"), http.StatusPaymentRequired)
+
+		v, err := parseVoucher(params)
+		if err != nil {
+			webError(w, fmt.Errorf("could not parse voucher: %w", err), http.StatusBadRequest)
 			return
 		}
-		rawChId := params.Get("channelId")
-
-		chId := types.Destination(common.HexToHash(rawChId))
 
-		// TODO: Allow this to be configurable
-		expectedPaymentAmount := big.NewInt(10)
+		// delta is the change in the channel balance caused by adding this voucher.
+		_, delta := h.nitroRpcClient.ReceiveVoucher(v)
 
-		hasPaid, err := checkPaymentChannelBalance(h.nitroRpcClient, chId, expectedPaymentAmount)
-		if err != nil {
-			webError(w, err, http.StatusPaymentRequired)
+		// TODO: A nil value indicates an error with the voucher. We should update to the latest go-nitro which properly returns the error.
+		if delta == nil {
+			webError(w, fmt.Errorf("invalid voucher received %+v", v), http.StatusBadRequest)
 			return
 		}
 
-		if !hasPaid {
-			webError(w, fmt.Errorf("payment of %d required", expectedPaymentAmount.Uint64()), http.StatusPaymentRequired)
+		// If the voucher resulted in a payment less than the expected payment, return an error.
+		if delta.Cmp(big.NewInt(expectedPayment)) < 0 {
+			webError(w, fmt.Errorf("payment of %d required, the voucher only resulted in a payment of %d", expectedPayment, delta.Uint64()), http.StatusPaymentRequired)
 			return
 		}
 	}
 
 	h.gwh.ServeHTTP(w, r)
 }
 
+// parseVoucher takes in an a collection of query params and parses out a voucher.
+func parseVoucher(params url.Values) (payments.Voucher, error) {
+	if !params.Has("channelId") {
+		return payments.Voucher{}, fmt.Errorf("a valid channel id must be provided")
+	}
+	if !params.Has("amount") {
+		return payments.Voucher{}, fmt.Errorf("a valid amount must be provided")
+	}
+	if !params.Has("signature") {
+		return payments.Voucher{}, fmt.Errorf("a valid signature must be provided")
+	}
+	rawChId := params.Get("channelId")
+	rawAmt := params.Get("amount")
+	amount := big.NewInt(0)
+	amount.SetString(rawAmt, 10)
+	rawSignature := params.Get("signature")
+
+	v := payments.Voucher{
```

Comment:
@geoknee brought up a good point about simplifying this by updating the go-nitro RPC client API to take multiple arguments(channelId,amount,signature) instead of a voucher struct. That way we wouldn't have to create a voucher here. 

I'll create an issue so we can consider this outside the scope of this PR.